File: HorseRacing.Contracts/Models/User/Responses/GetUser/GetUserResponse.cs
using HorseRacing.Contracts.Base.Responses;

namespace HorseRacing.Contracts.Models.User.Responses.GetUser
{
    public class GetUserResponse : BaseResponse<GetUserResponseDto>
    {
        public GetUserResponse() : base() { }
        public GetUserResponse(GetUserResponseDto data) : base(data) { }
    }
}

File: HorseRacing.Web/src/interfaces/api/contracts/model/user/responses/get-user/get-user-response.ts
import type { BaseResponse } from '~/interfaces/api/contracts/base/responses/base-response';
import type { GetUserResponseDto } from './get-user-response-dto';

export class GetUserResponse implements BaseResponse<GetUserResponseDto> {
    Data: GetUserResponseDto;

    constructor(data?: GetUserResponseDto) {
        this.Data = data || { FirstName: '', LastName: '', UserName: '', Phone: '', Email: '' };
    }
}
